### DISCUSSION TITLE
To put a task ON HOLD

### DISCUSSION THREAD

**Discussion by afsalabdup** [apache/airflow#24195] [Ideas]
Description: To put a future instance task ON HOLD.

There is a requirement where I want to put a task named 'task3' on hold, so that once the DAG is triggered as per schedule, all upstream tasks would start running but when it reaches 'task3', task3 should go "ON HOLD" state, so that downstream of task3 would not start running. If we can do that through CLI by providing future execution id then it would be great.

Currently I can mark a task to success through CLI by providing future execution id, but the issue is when I mark task3 to success, downstream tasks will start running immediately because downstream will check if its upstream task3 is success or not. I want to put task3 to ON HOLD so that it's downstream won't trigger by itself.

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by shamsuvs** [comment]
Is this feature available now?

**Comment by malthe** [comment]
And perhaps add optional operator option `email_on_hold` to be able to send an e-mail when a task is put on-hold.

**Comment by eladkal** [comment]
To my understanding a possible solution is to change the `On`/`Off` from DAG level to Task level.
So a DAG would be considered ON if at least one of it’s task is ON. Even assuming this is feasible it brings many challenges for example how would dag timeout behave if some tasks are on hold?

**Comment by malthe** [comment]
I think that's probably the best you can do yes. After each DAG run, use the `on_failure_callback` and `on_success_callback` methods to disable the DAG model (i.e., put it into "off" state).

**Comment by potiuk** [comment]
This is a nice feature- some kind of "manual approval" for a task. I think this is a good idea, but needs an AIP and devlist discussion. Hopefully someeone will pick it up. In the meantime I convert it to a discussion.
